### PR TITLE
Add TOC, some text fixes and rendering improvements

### PR DESCRIPTION
**Pull request by mmattel**
This PR adds a table of contents where appropriate and adds some rendering improvements.

We have done this in docs too which improves readability a lot.

Fixed broken xrefs and links.

The `\` before a URL helps finding broken links as it tells Antora to render it as text which is ok to example domains. The `\` it is an internal directive and will not be printed.

Live TOC example see: https://doc.owncloud.com/server/admin_manual/installation/system_requirements.html

### CODE REVIEW

**Comment by mmattel** [comment]
why is drone failing on non docs stuff?
this pr is docs only??

File: docs/modules/ROOT/pages/troubleshooting.adoc [hunk 1]
```diff
@@ -21,7 +25,7 @@ Ensure the WebDAV API is working::
...
   The ownCloud Client uses the built-in WebDAV access of the server content.
   Verify that you can log on to ownCloud's WebDAV server.
-  To verify connectivity with the ownCloud WebDAV server, open a browser window and enter the address to the ownCloud WebDAV server. For example, if your ownCloud instance is installed at `http://yourserver.com/owncloud`, your WebDAV server address isi `http://yourserver.com/owncloud/remote.php/webdav`. If you are prompted for your username and password but, after providing the correct credentials, authentication fails, please ensure that your authentication backend is configured properly.
+  To verify connectivity with the ownCloud WebDAV server, open a browser window and enter the address to the ownCloud WebDAV server. For example, if your ownCloud instance is installed at `\http://yourserver.com/owncloud`, your WebDAV server address isi `\http://yourserver.com/owncloud/remote.php/webdav`. If you are prompted for your username and password but, after providing the correct credentials, authentication fails, please ensure that your authentication backend is configured properly.
```
Comment (michaelstingl) [inline docs/modules/ROOT/pages/troubleshooting.adoc]:
```suggestion
  To verify connectivity with the ownCloud WebDAV server, open a browser window and enter the address to the ownCloud WebDAV server. For example, if your ownCloud instance is installed at `\http://yourserver.com/owncloud`, your WebDAV server address is `\http://yourserver.com/owncloud/remote.php/webdav`. If you are prompted for your username and password but, after providing the correct credentials, authentication fails, please ensure that your authentication backend is configured properly.
```

Comment (mmattel) [reply docs/modules/ROOT/pages/troubleshooting.adoc]: I fixed it. just to let you know, this type was already existant before my PR...

**Comment by ckamm** [comment]
Is the `master` target intentional? If this is intended to be included in 2.6.0 docs it should be merged into `2.6`.

File: docs/modules/ROOT/pages/troubleshooting.adoc [hunk 2]
```diff
@@ -146,7 +150,7 @@ To enable the writing of core dump files, you must define the `OWNCLOUD_CORE_DUM
 
 For example:
 
-`\` OWNCLOUD_CORE_DUMP=1 owncloud``
```
Comment (ckamm) [inline docs/modules/ROOT/pages/troubleshooting.adoc]: The call actually should be `OWNCLOUD_CORE_DUMP=1 owncloud`

Comment (mmattel) [reply docs/modules/ROOT/pages/troubleshooting.adoc]: Fixed

**Comment by mmattel** [comment]
@ckamm it is the master docs branch we are using, pls see the screenshot
image

**Comment by mmattel** [comment]
Ping ?

**Comment by ckamm** [comment]
@mmattel What will happen if you want to describe a feature in the master (to-be-2.7) docs that doesn't exist in 2.6? What if you want to fix documentation for 2.5 when the doc for 2.6 and master has significantly diverged since then? 

This is up to you, but I'd recommend against just using master. Having a branch per minor release makes sense and matches well to the different versions of doc that's released (https://doc.owncloud.org/desktop/2.5, https://doc.owncloud.org/desktop/2.6, ...)

**Comment by mmattel** [comment]
@ckamm thanks for your comment.
It is not my decision to use master or any other branch, this must be discussed and decided with @settermjd and @tboerger. I am just adding a PR to the master branch which is the branch currently used by docs. Just a view days ago, the 2.6 branch has been added as selectable choice in docs. This means, I follow with this PR the current way of pushing to master and will backport to 2.6 as soon this one is merged. Please note that this PR has no content change but adding a table of contents and some small fixes of existing content. I would like to ask for your approval and and your help in getting this merged. Many thanks for your support.

**Comment by tboerger** [comment]
The 2.6 branch had been enabled already in the meantime. The 2.5 docs are not updated anymore, they have not been migrated to the new process.

**Comment by ckamm** [comment]
I'll retarget to 2.6 and then merge to master.

**Comment by mmattel** [comment]
@ckamm thanks a lot

**Pull request merged**